substrate/bin/node-template/pallets/template/src/lib.rs

```diff
@@ -1,105 +1,188 @@
+	// this pallet.
 	#[pallet::pallet]
 	pub struct Pallet<T>(_);
 
-	/// Configure the pallet by specifying the parameters and types on which it depends.
+	/// The pallet's configuration trait.
+	///
+	/// All our types and constants a pallet depends on must be declared here.
+	/// These types are defined generically and made concrete when the pallet is declared in the
+	/// `runtime/src/lib.rs` file of your chain.
 	#[pallet::config]
 	pub trait Config: frame_system::Config {
-		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		/// The overarching runtime event type.
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
-		/// Type representing the weight of this pallet
+		/// A type representing the weights required by the dispatchables of this pallet.
 		type WeightInfo: WeightInfo;
 	}
 
-	// The pallet's runtime storage items.
-	// https://docs.substrate.io/main-docs/build/runtime-storage/
+	/// A storage item for this pallet.
+	///
+	/// In this template, we are declaring a storage item called `Something` that stores a single
+	/// `u32` value. Learn more about runtime storage here: <https://docs.substrate.io/build/runtime-storage/>
 	#[pallet::storage]
-	#[pallet::getter(fn something)]
-	// Learn more about declaring storage items:
-	// https://docs.substrate.io/main-docs/build/runtime-storage/#declaring-storage-items
 	pub type Something<T> = StorageValue<_, u32>;
 
-	// Pallets use events to inform users when important changes are made.
-	// https://docs.substrate.io/main-docs/build/events-errors/
+	/// Events that functions in this pallet can emit.
+	///
+	/// Events are a simple means of indicating to the outside world (such as dApps, chain explorers
+	/// or other users) that some notable update in the runtime has occurred. In a FRAME pallet, the
+	/// documentation for each event field and its parameters is added to a node's metadata so it
+	/// can be used by external interfaces or tools.
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
-		/// Event documentation should end with an array that provides descriptive names for event
-		/// parameters. [something, who]
-		SomethingStored { something: u32, who: T::AccountId },
+		/// A user has successfully set a new value.
+		SomethingStored {
+			/// The new value set.
+			something: u32,
+			/// The account who set the new value.
+			who: T::AccountId,
+		},
 	}
 
-	// Errors inform users that something went wrong.
+	/// Errors that can be returned by this pallet.
+	///
+	/// Errors tell users that something went wrong so it's important that they are informative.
+	/// Similar to events, error documentation is added to a node's metadata so it's important that
+	/// they have helpful documentation associated with them.
+	///
+	/// Note: the `Error` enum my always be generic over `<T>`, which is expanded to `<T: Config>`.
```

Comment:
```suggestion
	/// Note: the `Error` enum my always be generic over `<T>`, which is expanded to `<T: Config>`.
	/// (An `Error` can be up to 4 bytes in size should you want to return additional information.)
```

Reply:
This is useful information, but I'm not clear on what you're referring to - you mean that devs can also use `Other` from [`DispatchError`](https://paritytech.github.io/substrate/master/frame_support/dispatch/enum.DispatchError.html) and this can be up to 4 bytes ?

Reply:
@sacha-l I think it's probably referring to this: https://paritytech.github.io/substrate/master/src/sp_runtime/lib.rs.html#132

Reply:
Awesome thanks. Added in https://github.com/paritytech/polkadot-sdk/pull/1280/commits/74d300e0772f4d3e496ef8d8c45aac344f4dc57e